ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rebuild/NonNativeAcidMaterializedViewASTBuilder.java

```diff
@@ -39,6 +44,28 @@ public List<ASTNode> createDeleteSelectNodes(String tableName) {
             .collect(Collectors.toList()));
   }
 
+  @Override
+  public void appendDeleteSelectNodes(ASTNode selectNode, String tableName) {
+    Set<String> selectedColumns = new HashSet<>(selectNode.getChildCount());
+
+    for (int i = 0; i < selectNode.getChildCount(); ++i) {
+      ASTNode selectExpr = (ASTNode) selectNode.getChild(i);
+      ASTNode expression = (ASTNode) selectExpr.getChild(0);
+      if (expression.getType() == HiveParser.DOT &&
+          expression.getChildCount() == 2 &&
+          expression.getChild(0).getType() == HiveParser.TOK_TABLE_OR_COL) {
+        selectedColumns.add(expression.getChild(1).getText());
+      }
+    }
+
+    for (FieldSchema fieldSchema : mvTable.getStorageHandler().acidSelectColumns(mvTable, Context.Operation.DELETE)) {
```

Comment:
should we check for Null StorageHandler?

Reply:
It is checked earlier when `NonNativeAcidMaterializedViewASTBuilder` instance is created

https://github.com/apache/hive/blob/5e78ce0e1f1577c4b68f9149430c1d6ad26fd91d/ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rebuild/AlterMaterializedViewRebuildAnalyzer.java#L520-L521

https://github.com/apache/hive/blob/5e78ce0e1f1577c4b68f9149430c1d6ad26fd91d/ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java#L3388-L3391

Reply:
👍